### ISSUE TITLE
Use git submodule to include SWMM and EPANET engine projects

### ISSUE THREAD

**Issue by michaeltryby** [USEPA/SWMM-EPANET_User_Interface#79]
We should switch to including SWMM and EPANET engines in the UI project as submodules. This should help simplify the process of merging new engine versions with the UI. 

@TongZhai any thoughts?

**Comment by TongZhai** [comment]
@michaeltryby for running the simulation, we are already calling the DLL (which is a BIG module, technically speaking). So, we will need to think about what type of capability we would like to offer users by tapping directly into the inners of these models. Saving the entire state of the model inside the C modules is also likely, perhaps a bit unwieldy...

**Comment by michaeltryby** [comment]
Sorry @TongZhai I was not very clear. I'm only talking about changing repo organization. Read up on git [submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) command and I think you will understand what I am suggesting here.

**Comment by TongZhai** [comment]
@michaeltryby I don't see why not. So perhaps you can start by creating a new repository for the engines and we will go from there to setup the submodules inside this project.

**Comment by michaeltryby** [comment]
@TongZhai Yes and no. I have existing repos for engines [here](https://github.com/USEPA/Stormwater-Management-Model) and [here](https://github.com/USEPA/Water-Distribution-Network-Model). Submodules would allow us to include them directly in this project. There is no need to create new repos. 

There is a way to create a folder within this repo that is a submodule linked back to the engine repo. I will read up on it over the weekend.
